### DISCUSSION TITLE
Why ignore Exception when Compile Binding?

### DISCUSSION THREAD

**Discussion by ly3027929699** [AvaloniaUI/Avalonia#13889] [Q&A] [answered]
![image](https://github.com/AvaloniaUI/Avalonia/assets/42492716/37c2a7e7-a7fc-44f8-adff-ead30d4f9826)
I must catch the exception at each place instead of 'AppDomain.CurrentDomain.UnhandledException'
![image](https://github.com/AvaloniaUI/Avalonia/assets/42492716/a8b8703c-c598-455c-814a-719da2c4f51d)

**Comment by ly3027929699** [comment]
Is there any other way to catch the unhandled exception?

**Comment by maxkatz6** [comment]
https://docs.avaloniaui.net/docs/guides/development-guides/data-validation

**Reply by timunie** [reply, marked as answer]
the validation only works in the setter iirc, not in the getter part. Also Microsoft says that throwing inside a getter of a property should be avoided, see: https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/property?redirectedfrom=MSDN

**Reply by ly3027929699** [reply]
Thanks for your replies.
 I hope it can show all errors of binding in the developing time because I have no WPF experience. I prefer to throw the error for a huge tip to locate the bugs

**Reply by timunie** [reply]
CompiledBindings help a lot: https://docs.avaloniaui.net/docs/basics/data/data-binding/compiled-bindings

More over maybe check out this: https://docs.avaloniaui.net/docs/concepts/unhandledexceptions

**Reply by ly3027929699** [reply]
thanks, I remove the try{...}catch{} on my side to throw the error
